### PR TITLE
8.x-3.0-beta1

### PR DESCRIPTION
**Pull request by hstandaert**
# CHANGELOG
## Added

* DTGB-491: Added [`dg_maps`](https://github.com/digipolisgent/drupal_module_dg-maps) theming
* Tabs/local tasks theming
    > :warning: The tabs block should ideally be placed in the admin region. 

## Removed

* DTGB-683: Frequently visited.
* DTGB-700: [Styleguide](https://github.com/StadGent/fractal_styleguide_gent-base) as a part of gent-base. It is integrated as an npm dependency now.

## Fixed

* DTGB-706: Fixed missing document link text when no file description is set.

# How to upgrade

## Remove `styleguide` references

Since we use source files from the styleguide to build our own assets, we'll have to update some paths in our custom themes as well. More specifically, the themes that are based on the starterkit. 

To make this easier, you can download+execute a remote script that replaces the paths for you. It only needs to know the path to your custom theme.

1. `cd` into the root directory of your Drupal project
2. Set the `THEME_FOLDER` variable to the path where your custom theme is located. Make it relative to you current directory.

    ```bash
    export THEME_FOLDER=./web/themes/custom/{your_theme}
    ```

3. Execute the remote script to find & replace the references to the removed `styleguide` folder

    ```bash
    wget https://git.io/fhxXh -O - | sh
    ```

## Update nvm version in travis. 

The node version that `styleguide` requires is upgraded to `>=10.14.2` and the trusty build environment doesn't have this version pre-installed.

If you encounter any issues with this, add: 

```yml
before_install:
  - nvm install 10.14.2 && nvm use 10.14.2
```

### CODE REVIEW

File: gent_base.theme [hunk 1]
```diff
@@ -620,8 +633,7 @@ function gent_base_preprocess_paragraph__timeline_item__alt(&$variables) {
 /**
  * Implements hook_preprocess_HOOK().
```
Comment (zero2one) [inline gent_base.theme]: Implements hook_preprocess_paragraph__contact().

File: gent_base.theme [hunk 2]
```diff
@@ -925,3 +937,13 @@ function gent_base_preprocess_vesta(&$variables) {
     }
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
```
Comment (zero2one) [inline gent_base.theme]: Implements hook_preprocess_dg_maps_map().

File: templates/contrib/dg_maps/dg-maps-map-region.html.twig [hunk 3]
```diff
@@ -0,0 +1,20 @@
+{% set region_classes = [
+'map-region',
+'map-region--' ~ name,
```
Comment (zero2one) [inline templates/contrib/dg_maps/dg-maps-map-region.html.twig]: If no name is present a class name `map-region--` will be added.

File: templates/contrib/dg_maps/dg-maps-map.html.twig [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{{ attach_library('gent_base/maps') }}
+{{ attach_library('gent_base/tabs') }}
+
```
Comment (zero2one) [inline templates/contrib/dg_maps/dg-maps-map.html.twig:3]: Single empty line.

File: templates/custom/vesta/vesta.html.twig [hunk 5]
```diff
@@ -162,4 +162,3 @@
   </div>
 </aside>
 {% endif %}
-
```
Comment (zero2one) [inline templates/custom/vesta/vesta.html.twig]: Why is the (required) empty line on the end of the file removed?

File: templates/contrib/facets/facets-summary-item-list.html.twig [hunk 6]
```diff
@@ -66,4 +66,3 @@
     {{- empty -}}
   {%- endif -%}
 {%- endif %}
-
```
Comment (zero2one) [inline templates/contrib/facets/facets-summary-item-list.html.twig]: Why is the (required) empty line on the end of the file removed?

File: source/js/dg_maps.baselayerswitcher.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
+/* global ol */
+/**
+ * @file
+ * DG Maps functionality extensions.
+ */
+(function ($, Drupal) {
+  'use strict';
+
+  var originalPrototype = Drupal.dgMaps.ol.control.BaseLayerSwitcher.prototype;
+
```
Comment (zero2one) [inline source/js/dg_maps.baselayerswitcher.js:10]: Remove empty line (there can be only one).

File: source/js/dg_maps.baselayerswitcher.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+  /**
+   * Base Layer Switcher control.
+   *
+   * @constructor
+   * @extends {ol.control.Control}
+   * @param {Object=} opt_options Control options.
+   */
+  Drupal.dgMaps.ol.control.BaseLayerSwitcher = function (opt_options) {
+
```
Comment (zero2one) [inline source/js/dg_maps.baselayerswitcher.js]: Remove empty line.

File: source/js/dg_maps.bindings.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  'use strict';
+
+  Drupal.behaviors.gentBaseLoadDGMaps = {
+    attach: function (context, settings) {
+      if (!allieTabs) { // eslint-disable-line no-undef
+        return;
+      }
+
+      var selected = document.querySelectorAll('.js-map-tabs');
+      for (var s = selected.length; s--;) {
```
Comment (zero2one) [inline source/js/dg_maps.bindings.js:17]: A lot of nesting going on here...
Better to put sub-nesting in separate functions.

**Comment by zero2one** [comment]
There is a lot of javascript code to alter/add the dg_maps functionality. This should be added to the dg_maps module instead so all projects can use this functionality.

**Comment by hstandaert** [comment]
> There is a lot of javascript code to alter/add the dg_maps functionality. This should be added to the dg_maps module instead so all projects can use this functionality.

@zero2one, these are strictly styling overrides (since a lot of the markup is built in javascript in the dg_maps module), so I don't think it's useful for non-styleguide projects.

File: CHANGELOG.md [hunk 10]
```diff
@@ -2,6 +2,23 @@
 
 All Notable changes to `digipolisgent/drupal_theme_gent-base`.
 
+## [8.x-3.0-beta1]
+
+### Added
+
+* DTGB-491: Added [`dg_maps`](https://github.com/digipolisgent/drupal_module_dg-maps) theming
```
Comment (zero2one) [inline CHANGELOG.md]: Sentenced should end with a full stop (.).

File: CHANGELOG.md [hunk 11]
```diff
@@ -2,6 +2,23 @@
...
+* Tabs/local tasks theming
+    > :warning: The tabs block should ideally be placed in the admin region. 
+
+### Removed
+
+* DTGB-683: Frequently visited.
+* DTGB-700: [Styleguide](https://github.com/StadGent/fractal_styleguide_gent-base) as a part of gent-base. It is integrated as an npm dependency now.
```
Comment (zero2one) [inline CHANGELOG.md]: Lines should not be longer than 80 characters.

**Review by zero2one — CHANGES_REQUESTED** [review]

**Review by zero2one — APPROVED** [review]

**Pull request merged**